src/xunit.v3.assert.tests/Asserts/Sdk/ArgumentFormatterTests.cs

```diff
@@ -120,6 +120,18 @@ public static void CharacterValue(char value, string expected)
 			Assert.Equal(expected, ArgumentFormatter.Format(value));
 		}
 
+		[CulturedFact]
+		public static void FloatValue()
+		{
+			Assert.Equal(Math.PI.ToString("G7"), ArgumentFormatter.Format((float)Math.PI));
```

Comment:
`G7` isn't enough digits for `float`, you need at least `9` digits to guarantee uniqueness
Likewise, `G19` is more precise than necessary for `double`, you only need `17` to guarantee uniqueness

`R` will generally be unique on .NET Framework (outside bugs) and will always be unique on .NET Core 3.0+